### PR TITLE
Add metrics support

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Add support for Agent metrics and Gateway metrics.
- Support metrics merging
- Add UI options for metrics
- Add demo templates for Prometheus and Grafana

How I've tested this PR: These PRs have been tested in composite PRs.

How I expect reviewers to test this PR: Verify CHANGELOG changes

Checklist:
- [x] Bats tests added
- [x] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 BREAKING CHANGES:
 
 FEATURES:
+* Metrics: add support for metrics in Consul. This enables support for Consul Agent metrics,
+  Consul Gateway metrics, metrics merging to serve both application and sidecar metrics and support to configure a metrics provider for the Consul UI.
+  Additionally, adds templates for a demo installation of Prometheus and Grafana.
```
Comment (ndhanushkodi) [inline CHANGELOG.md:8]: Can we add that metrics merging will only work with Consul 1.10.0-alpha1 and higher?

**Review by ndhanushkodi — APPROVED** [review]
🎉

File: CHANGELOG.md (hunk 1 above)
Comment (thisisnotashwin) [reply CHANGELOG.md:8]: Made the change.

**Review by lkysow — APPROVED** [review]
📈

**Comment by lkysow** [comment]
Reminder that you may want to rebase merge this to retain your commits as is since they look well scoped.

**Pull request merged**